x/tss/keeper/grpc_query.go

```diff
@@ -58,3 +62,71 @@ func (q Querier) AssignableKey(c context.Context, req *types.AssignableKeyReques
 
 	return &types.AssignableKeyResponse{Assignable: !assigned}, nil
 }
+
+// ValidatorKey returns a map of active multisig role key ids to pub keys of a validator
+func (q Querier) ValidatorKey(c context.Context, req *types.ValidatorKeyRequest) (*types.ValidatorKeyResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	valAddress, err := sdk.ValAddressFromBech32(req.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := q.staker.Validator(ctx, valAddress)
+	if validator == nil {
+		return nil, fmt.Errorf("not a validator")
+	}
+
+	chains := q.nexus.GetChains(ctx)
+	keys := make(map[string]*types.ValidatorKeyResponse_Keys, 10)
+
+	keyRoles := []exported.KeyRole{exported.MasterKey, exported.SecondaryKey}
+
+	for _, chain := range chains {
+		for _, keyRole := range keyRoles {
+			if currentKey, found := q.keeper.GetCurrentKey(ctx, chain, keyRole); found {
+				if valKeys, ok := q.getSerializedMultisigKeys(ctx, currentKey, valAddress); ok {
```

Comment:
This would be problematic if TSS is back. The gRPC method should be renamed to `ValidatorMultisigKeys` and key type should be checked here and skipped instead of erroring if the key type isn't multisig.

Reply:
Yeah, renamed. I do skip the key id if the key type isn't multisig in `getSerializedMultisigKeys`. I added the check there to reduce code duplication.